### PR TITLE
update Microservice package

### PR DESCRIPTION
**Pull request by dmitriyslaym**
Related to #75

### CODE REVIEW

File: packages/scalecube-microservice/src/api/Microservice.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
 import { ProxyOptions, ServiceCall, CreateServiceCallOptions } from '.';
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/Microservice.ts]:
```
/**
 * Microservice - provide functionality of microservice instance.
 *
 * `destroy` - The method is used to delete microservice and close all the subscriptions related with it
 *
 * `createProxy<T>: T` -  Create a proxy to a method and allowe extra logic when invoked.
 *
 * `createServiceCall` -  Expose serviceCall to use (not via Proxy)
 *
 */
```

File: packages/scalecube-microservice/src/api/MicroserviceOptions.ts [hunk 2]
```diff
@@ -1,5 +1,11 @@
 import { Service } from '.';
 
+/**
+ * The options for the creation of microservice container
+ */
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/MicroserviceOptions.ts]: /**
 * MicroserviceOptions - The options for the creation of microservice instance.
 *
 * @param `services` - An array of services
 *
 */

File: packages/scalecube-microservice/src/api/Microservices.ts [hunk 3]
```diff
@@ -1,5 +1,11 @@
 import { MicroserviceOptions, Microservice } from '.';
 
+/**
+ * The factory for the creation of microservices containers
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/Microservices.ts]:
```
/**
 * Microservices - The factory for the creation of microservices containers.
 *
 * `create` - Create microservice instance
 *
 */
```

File: packages/scalecube-microservice/src/api/ProxyOptions.ts [hunk 4]
```diff
@@ -1,6 +1,16 @@
 import { ServiceDefinition, Router } from '.';
 
+/**
+ * The options that are used for the creation of the proxy for the specific microservice container
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/ProxyOptions.ts]:
```
/**
 * ProxyOptions - The options that are used for the creation of the proxy for the specific microservice container
 *
 * @param router - Router logic
 *
 * @param serviceDefinition - Service definition for the required service
 *
 */
```

File: packages/scalecube-microservice/src/api/Reference.ts [hunk 5]
```diff
@@ -1,11 +1,29 @@
 import AsyncModel from './AsyncModel';
 
+/**
+ * The reference wrapper of the implementation of the method from a service
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/Reference.ts]:
```
/**
 * Reference - Define local service data
 *
 * @param qualifier - The combination of serviceName and methodName: serviceName/methodName
 * 
 * @param serviceName - The name of a service that is provided in serviceDefinition
 * 
 * @param methodName - The name of a method that is provided in the serviceDefinition
 * 
 * @param asyncModel - Type of communication between a consumer and a provider
 * 
 * @param reference - mapping methodName and implementation
 * 
 */
```

File: packages/scalecube-microservice/src/api/RouteOptions.ts [hunk 6]
```diff
@@ -1,6 +1,15 @@
 import { LookUp, Message } from '.';
 
+/**
+ * The options that are used for the creation of the custom router
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/RouteOptions.ts]: /**
 * RouteOptions - The options that are used for the creation of the custom router
 *
 * @param lookUp - Functionality to find removeService by given critiria
 * 
 * @param message - Data that is used for executing the call of service method
 * 
 */

File: packages/scalecube-microservice/src/api/Router.ts [hunk 7]
```diff
@@ -1,5 +1,12 @@
 import { Endpoint, RouteOptions } from '.';
 
+/**
+ * Allows to specify the logic of which instance of the required endpoint should be returned for the future call
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/Router.ts]:
```
/**
 * Router - a way to Specify logic for the pick a remoteService
 *
 * `route` - logic to pick remoteService
 */
```

File: packages/scalecube-microservice/src/api/Service.ts [hunk 8]
```diff
@@ -1,6 +1,16 @@
 import { ServiceImplementation, ServiceDefinition } from '.';
 
+/**
+ * The information about service and the implementation for all the methods that are included in it
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/Service.ts]:
```
/**
 * Service - definition and implementation of a service
 *
 * @param definition - Metadata of a service.
 * 
 *  Includes the name of a service and the map of methods that are included in it.
 *
 * @param reference - Implementation of a service.
 * 
 */
```

File: packages/scalecube-microservice/src/api/ServiceCall.ts [hunk 9]
```diff
@@ -1,7 +1,18 @@
 import { Observable } from 'rxjs';
 import { Message } from '.';
 
+/**
+ * Provides an ability to run the method from a microservice container for a specific async model
+ */
```
Comment (IdoJS) [inline packages/scalecube-microservice/src/api/ServiceCall.ts]:
```
/**
 * ServiceCall - Provides serviceCall functionality by asyncModel
 *
 * @param requestStream - Observable serviceCall
 *
 * @param requestResponse - Promise serviceCall
 * 
 */
```

File: packages/scalecube-microservice/src/api/AsyncModel.ts [hunk 10]
```diff
@@ -1,7 +1,17 @@
+/**
+ * "Request Stream" type of async model - a consumer requires a stream and receives the emits from a provider after
+ * the subscription
+ */
 export type RequestStreamAsyncModel = 'RequestStream';
```
[the remark this answers is not in the document]
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/AsyncModel.ts]: agree

File: packages/scalecube-microservice/src/api/AsyncModel.ts [hunk 11]
```diff
@@ -1,7 +1,17 @@
...
 
+/**
+ * "Request Response" type of async model - a consumer requires data and a provider responds with the data once
+ */
 export type RequestResponseAsyncModel = 'RequestResponse';
```
[the remark this answers is not in the document]
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/AsyncModel.ts]: agree

File: packages/scalecube-microservice/src/api/CreateServiceCallOptions.ts [hunk 12]
```diff
@@ -1,5 +1,11 @@
 import { Router } from '.';
 
+/**
+ * Options for the creation of Service Call
```
[the remark this answers is not in the document]
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/CreateServiceCallOptions.ts]: Custom router helps to specify the logic of choosing the appropriate remoteService

File: packages/scalecube-microservice/src/api/Endpoint.ts [hunk 13]
```diff
@@ -1,10 +1,33 @@
 import AsyncModel from './AsyncModel';
 
+/**
+ * The endpoint to the part of the functionality of a service, that provides a user with an opportunity to run this
+ * peace of functionality
```
[the remark this answers is not in the document]
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/Endpoint.ts]: Defines remote service data

File: packages/scalecube-microservice/src/api/LookUp.ts [hunk 14]
```diff
@@ -1,3 +1,6 @@
 import { Endpoint, LookupOptions } from '.';
 
+/**
+ * The function that provides all the appropriate endpoints for a given search criteria
```
[the remark this answers is not in the document]
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/LookUp.ts]: The function that finds all the appropriate endpoints for a given search criteria

File: packages/scalecube-microservice/src/api/Message.ts [hunk 15]
```diff
@@ -1,4 +1,13 @@
+/**
+ * Data that is used for executing the call of service method
+ */
```
[the remark this answers is not in the document]
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/Message.ts]: agree

File: packages/scalecube-microservice/src/api/Microservice.ts (hunk 1 above)
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/Microservice.ts]: provides functionality of microservice container

Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/Microservice.ts]: Creates a proxy to a method and provides extra logic when invoked

Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/Microservice.ts]: Exposes serviceCall to a user (not via Proxy)

File: packages/scalecube-microservice/src/api/ProxyOptions.ts (hunk 4 above)
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/ProxyOptions.ts]: Custom router specifies the logic of choosing the appropriate remoteService

File: packages/scalecube-microservice/src/api/Reference.ts (hunk 5 above)
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/Reference.ts]: Define local service data

File: packages/scalecube-microservice/src/api/Router.ts (hunk 7 above)
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/Router.ts]: The method that returns the appropriate remoteService for the provided search criteria. Returns null if no
appropriate remoteService was found

File: packages/scalecube-microservice/src/api/Service.ts (hunk 8 above)
Comment (dmitriyslaym) [reply packages/scalecube-microservice/src/api/Service.ts]: definition and implementation of a service

**Pull request merged**